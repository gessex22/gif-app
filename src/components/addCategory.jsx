import { useState } from "react";

export const AddCategory = ({ onNewCategory }) => {
  const [imputValue, setImputValue] = useState("one punch");

  const onImputChange = (event) => {
    setImputValue(event.target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    if (imputValue.trim().length <= 1) return;
    onNewCategory(imputValue.trim());
    setImputValue("");
  };

  return (
    <form onSubmit={onSubmit}>
      <input
        type="text"
        placeholder="Buscar"
        value={imputValue}
        onChange={onImputChange} // (event) => onImputChange(event)
      />
    </form>
  );
};
