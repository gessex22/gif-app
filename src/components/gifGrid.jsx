export const GifGrid = ({category}) =>{

    const gifs = [1,2,3,4,4,6]

    return  (
    <>
        <h3> {category}</h3>
      {
        gifs.map( gif=> 
            <p> {gif} </p>
        )
      }
    </>
    )
}