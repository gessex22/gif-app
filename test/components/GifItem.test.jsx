import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Pruebas en <GifItem>", () => {
  const title = "saitama";
  const url = "https://mrrobot.com/image/background.jpg";
  test("Debe hacer match snapchot", () => {
    const { container } = render(<GifItem  title={ title} url={url}/>);
    expect(container).toMatchSnapshot()

  });


  test('Debe de mostrar la imagen con el URL y el ALT indicado', ()=>{
    render(<GifItem title={title} url={url} />)
    const {alt,src} = screen.getByRole("img")
    expect(alt).toBe( title)
    expect(src).toBe( url)
  })

  test('Debe de mostrar el titulo del componente ', ()=>{
    render(<GifItem  title={title} url={url } />)
    //const {text} = screen.getByText(title).toBeTruthy()
    expect( screen.getByText(title)).toBeTruthy()
  })
});
