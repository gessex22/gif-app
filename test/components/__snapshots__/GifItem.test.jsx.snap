// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem> Debe hacer match snapchot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="saitama"
      src="https://mrrobot.com/image/background.jpg"
    />
    <p>
       
      saitama
    </p>
  </div>
</div>
`;
